Jenkinsfile

```diff
@@ -155,79 +155,23 @@ pipeline {
             echo 'Skipping Kafka Streams archetype test for Java 17'
           }
         }
-        
-        // To avoid excessive Jenkins resource usage, we only run the stages
-        // above at the PR stage. The ones below are executed after changes
-        // are pushed to trunk and/or release branches. We achieve this via
-        // the `when` clause.
-        
-        stage('JDK 8 and Scala 2.13') {
```

Comment:
Adding this note here for future visitors to this PR (and to ensure that we are on the same page here about the reason to remove these stages from CI)

Removing this combination from tests in fine because scala 2.13 is fully compatible with JDK 8 [1]. If we can guarantee that code is compiling correctly with 2.13 and JDK 8 separately, then that is sufficient and we do not necessarily have to test a unique combination of both.

[1] https://docs.scala-lang.org/overviews/jdk-compatibility/overview.html

Reply:
That statement isn't strictly true since there could be bugs in the implementation - that's why we had these tests in the first place. But the truth is that they're low value since (1) the probability of a bug affecting JDK 8 and Scala 2.13 but not JDK 8 and Scala 2.12 is low (2) having too many variants makes it even less likely for people to pay attention to the flaky failures that exist (3) it's expensive to run so many variants.